vertx-web/src/main/java/io/vertx/ext/web/handler/impl/SessionHandlerImpl.java

```diff
@@ -215,11 +223,13 @@ private void addStoreSessionHandler(RoutingContext context) {
   private void createNewSession(RoutingContext context) {
     Session session = sessionStore.createSession(sessionTimeout, minLength);
     context.setSession(session);
-    Cookie cookie = Cookie.cookie(sessionCookieName, session.id());
-    cookie.setPath("/");
-    cookie.setSecure(sessionCookieSecure);
-    cookie.setHttpOnly(sessionCookieHttpOnly);
-    // Don't set max age - it's a session cookie
+    Cookie cookie = Cookie.cookie(sessionCookieName, session.id())
+      .setPath("/")
+      .setSecure(sessionCookieSecure)
+      .setHttpOnly(sessionCookieHttpOnly);
+    // set max age if user requested it - else it's a session cookie
+    if(sessionCookieMaxAge > 0)
```

Comment:
From a security perspective I'm not sure if this is corrent, this means that sessions can be hijacked if a user closes the browser and assumes it is closed and a bad user just reopens the browser and keeps browsing...